src/net/net.c

```diff
@@ -83,14 +80,18 @@ void *lwip_allocate(u64 size)
     /* To maintain the malloc/free interface with mcache, allocations must stay
        within the range of objcaches and not fall back to parent allocs. */
     assert(size <= U64_FROM_BIT(MAX_LWIP_ALLOC_ORDER));
-    void *p = allocate_zero(lwip_heap, size);
+    spin_lock(&lwip_heap_lock);
```

Comment:
Maybe this wouldn't cause a problem as it is called from lwip_init() and before lwip is in use, but the closure allocation from lwip_heap in sys_timeouts_init() (as well as the closure finish, if it's ever actually invoked) should be protected by lwip_heap_lock as well. Using a locking_heap_wrapper in lieu of manually locking would eliminate such cases of unlocked use.